include/staggered_kd_build_xinv.h

```diff
@@ -13,17 +15,21 @@ namespace quda
   */
   void BuildStaggeredKahlerDiracInverse(GaugeField &Xinv, const cudaGaugeField &gauge, const double mass);
 
+  /**
+     @brief Perform the reordering of the Kahler-Dirac inverse block from a coarse scalar field to a KD geometry gauge field
+     @param[out] out Kahler-Dirac inverse in KD geometry gauge field
+     @param[in] in Kahler-Dirac inverse in coarse geometry MILC layout
+     @param[in] dagger_approximation whether or not we're doing the dagger approximation, where you pass in X instead
+     @param[in] mass mass of the original staggered operator w/out factor of 2 convention, needed for dagger approx
+  */
+  void ReorderStaggeredKahlerDiracInverse(GaugeField &xInvFineLayout, const GaugeField &xInvCoarseLayout, const bool dagger_approximation, const double mass);
+
   /**
      @brief Allocate and build the Kahler-Dirac inverse block for KD operators
      @param[in] in gauge original fine gauge field
      @param[in] in mass the mass of the original staggered operator w/out factor of 2 convention
-     @param[in] in precision of Xinv field
-     @return constructed Xinv, which needs to be deleted manually
+     @return constructed Xinv
   */
-  cudaGaugeField *AllocateAndBuildStaggeredKahlerDiracInverse(const cudaGaugeField &gauge, const double mass,
-                                                              const QudaPrecision override_prec);
+  std::unique_ptr<GaugeField> AllocateAndBuildStaggeredKahlerDiracInverse(const cudaGaugeField &gauge, const double mass);
```

Comment:
I assume not, but just checking that this header is not included in device code?  The presence of smart pointers will break Jitify,

Reply:
Can confirm, I've run this PR through jitify (well, at least a version after I made this update) and there are no issues related to this file.